core/primitives/src/receipt.rs

```diff
@@ -91,25 +299,44 @@ impl Receipt {
     /// receipt_id. It contains `signer_id` and `signer_public_key` to indicate this is a gas
     /// refund. The execution of this receipt will try to refund the allowance of the
     /// access key with the given public key.
+    /// Gas refund does not inherit priority from its parent receipt and has no priority associated with it
     /// NOTE: The access key may be replaced by the owner, so the execution can't rely that the
     /// access key is the same and it should use best effort for the refund.
     pub fn new_gas_refund(
         receiver_id: &AccountId,
         refund: Balance,
         signer_public_key: PublicKey,
+        priority: ReceiptPriority,
     ) -> Self {
-        Receipt {
-            predecessor_id: "system".parse().unwrap(),
-            receiver_id: receiver_id.clone(),
-            receipt_id: CryptoHash::default(),
+        match priority {
+            ReceiptPriority::Priority(priority) => Receipt::V1(ReceiptV1 {
+                predecessor_id: "system".parse().unwrap(),
+                receiver_id: receiver_id.clone(),
+                receipt_id: CryptoHash::default(),
 
-            receipt: ReceiptEnum::Action(ActionReceipt {
-                signer_id: receiver_id.clone(),
-                signer_public_key,
-                gas_price: 0,
-                output_data_receivers: vec![],
-                input_data_ids: vec![],
-                actions: vec![Action::Transfer(TransferAction { deposit: refund })],
+                receipt: ReceiptEnum::Action(ActionReceipt {
+                    signer_id: receiver_id.clone(),
+                    signer_public_key,
+                    gas_price: 0,
+                    output_data_receivers: vec![],
+                    input_data_ids: vec![],
+                    actions: vec![Action::Transfer(TransferAction { deposit: refund })],
+                }),
+                priority,
+            }),
```

Comment:
Yeah I don't want to introduce a new protocol version in this PR (because it is supposed to be a no-op) so I didn't go with that approach. I can also change this code to only generate V0 for now if it is confusing